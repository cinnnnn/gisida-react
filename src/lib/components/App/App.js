import React, { Component } from 'react';
import { SupAuth } from 'gisida';
import './App.scss';

class App extends Component {
  constructor() {
    super()
    this.checkIfAuthenticated = this.checkIfAuthenticated.bind(this)
  }

  checkIfAuthenticated() {
    const isAuthenticated = JSON.parse(localStorage.getItem('AUTH')).AUTH.isAuthenticated
    const csvId = JSON.parse(localStorage.getItem('csvId'))
    const token = localStorage.getItem('access_token')
    if (csvId && isAuthenticated) {
      SupAuth.getMediaAuthConfig(csvId, token)
    }
    
  }

  render() {
    this.checkIfAuthenticated()
    return (
      <div className="app">
        {this.props.children}
      </div>
    );
  }
}

export default App;
