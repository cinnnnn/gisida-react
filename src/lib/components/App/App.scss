body { 
  margin: 0; 
  padding: 0; 
}

.app {
  font-family: 'Montserrat', sans-serif;
}
