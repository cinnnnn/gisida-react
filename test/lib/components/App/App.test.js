import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import App from '../../../../src/lib/components/App/App';

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    
  });

  it('App renders child components', () => {
    const component = renderer.create(<App />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
