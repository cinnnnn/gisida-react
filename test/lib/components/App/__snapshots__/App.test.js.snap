// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App renders child components 1`] = `
<div
  className="app"
/>
`;
